lib/payment_processor/clients/braintree/subscription.rb

```diff
@@ -6,6 +6,13 @@ def self.make_subscription(amount:, plan_id:, payment_method_token:)
           new(amount, plan_id, payment_method_token).subscribe
         end
 
+        def self.make_subscription_from_transaction(transaction)
+          amount = transaction...
+          plan_id = ENV['BRAINTREE_SUBSCRIPTION_PLAN_ID']
+          payment_method_token = transaction...
+          new(amount, plan_id, payment_method_token).subscribe
+        end
+
```

Comment:
I've tried testing this against braintree's API. It doesn't seem to like us making a subscription immediately after a sale! The transaction goes through, but the subscription returns a 422. Your comment above makes perfect sense, I just wish I'd seen it yesterday before checking out!:

> Hey - I was thinking about this - wouldn't it work to just make a subscription with an immediate payment if the user wants to subscribe? When does the subscription get charged otherwise?

So I guess the best way to handle a recurring donation is to first create a customer with their payment nonce, (this gives us the required payment method token) and then the subscription, setting the billing date to the date of the request:

https://developers.braintreepayments.com/reference/request/subscription/create/ruby#first_billing_date

Thoughts?